1.8.3/amd64/alpine/entrypoint.sh

```diff
@@ -37,6 +37,43 @@ case ${OPENHAB_VERSION} in
         cp -av "${APPDIR}/userdata.dist/." "${APPDIR}/userdata/"
       fi
 
+      # Upgrade userdata if versions do not match
+      curVersion=$(< ${APPDIR}/userdata/etc/version.properties grep build-no | cut -d : -f 2 | tr -d '[:space]')
+      imgVersion=$(< ${APPDIR}/userdata.dist/etc/version.properties grep build-no | cut -d : -f 2 | tr -d '[:space]')
+      echo "Current \"${curVersion}\" Image \"${imgVersion}\"" # TODO remove
+      if [ "${curVersion}" != "${imgVersion}" ]; then
+        echo "Image version \"${imgVersion}\" is different from userdata version \"${curVersion}\""
+
+        # Make a backup of userdata
+        backupFile=userdata-$(date +"%FT%H:%M:%S").tar
+        tar cf /tmp/${backupFile} ${APPDIR}/userdata
+        if [ ! -d "${APPDIR}/userdata/backup" ]; then
+          mkdir ${APPDIR}/userdata/backup
+        fi
+        cp /tmp/${backupFile} ${APPDIR}/userdata/backup/
+        echo "You can find backup of userdata in ${APPDIR}/userdata/backup/${backupFile}"
+
+	# Copy over the updated files
```

Comment:
Do we really want to also be managing this list here? Would it be better to call the backup and update scripts instead?

Reply:
The update script will download the zip file each time. Would it be better to include the list of files in the distribution for a package manager (and the backup/restore scripts) to use?

Reply:
Yes that would be great :-)

Reply:
No, I don't really but it was a bigger job across two repos to pull them out into a list that can  be used in both places. This list doesn't change that frequently so I figured we would kick the can down the road just a bit until we can figure out how to make the list usable for both.

My initial thoughts were perhaps that part of the script could be pulled out into a separate script that both entrypoint and all the installation and upgrade scripts call. Then not only would the list be reusable here but the actual behavior. That would guarantee that Docker would keep up with changes to the upgrade process in the future. I don't know if the apt/yum installers can support that though.

Reply:
For me it would be okay as a first step.

Btw: I think that @BClark09 did not mean it as an improvement on this PR but the openHAB-generic side of things.

Reply:
Yep that's right, I agree what you're doing is good for now but when I have the time (may be a few weeks), I'll add the proposed list to `openhab-distro`, and make subsequent PRs for `openhab-docker` and `openhab-linuxpkg`. :)